compiler-rt/test/nsan/vec_sqrt.cpp

```diff
@@ -0,0 +1,35 @@
+// RUN: %clangxx_nsan -O0 -g -mavx %s -o %t
+// RUN: NSAN_OPTIONS=check_nan=true,halt_on_error=0 %run %t 2>&1 | FileCheck %s
+// RUN: %clangxx_nsan -O3 -g -mavx %s -o %t
+// RUN: NSAN_OPTIONS=check_nan=true,halt_on_error=0 %run %t 2>&1 | FileCheck %s
+
+#include <cmath>
+#include <immintrin.h>
+#include <iostream>
+
+void simd_sqrt(const float *input, float *output, size_t size) {
+  size_t i = 0;
+  for (; i + 7 < size; i += 8) {
+    __m256 vec = _mm256_loadu_ps(&input[i]);
+    __m256 result = _mm256_sqrt_ps(vec);
+    _mm256_storeu_ps(&output[i], result);
+    // when simd is used, the warning is not triggered
```

Comment:
it looks like something is broken.
we'd like to detect cases like this, the report below  triggered by std::sqrt below is not relevant, need to figure out/understand what's broken for simd.

Reply:
Sure, I’ll check the generated LLVM IR to identify where the issue might be.

Reply:
I've simplified the test a bit:
```
#include <cassert>
#include <cmath>
#include <immintrin.h>
#include <iostream>

void simd_sqrt(const float *input, float *output, size_t size) {
  size_t i = 0;
  assert(size % 8 == 0);
  for (; i < size; i += 8) {
    __m256 vec = _mm256_loadu_ps(&input[i]);
    __m256 result = _mm256_sqrt_ps(vec);
    _mm256_storeu_ps(&output[i], result);
  }
}

int main() {
  float input[] = {0.0, 1.0, -2.0, 3.0, 4.0, 5.0, 6.0, 7.0};
  float output[8];

  simd_sqrt(input, output, 8);

  for (int i = 0; i < 8; ++i) {
    std::cout << output[i] << std::endl;
  }
  return 0;
}
```

```
clang++ -fsanitize=numerical -mavx -O3 -g ~/simd2.cpp -o ~/simd2.exe
```

It also works as expected.

Reply:
When using `sqrt`, it detects `nan`, but I'm not sure if using `_mm256_sqrt_ps` would exhibit similar behavior.